File: Java/Trees/Model/BinaryTree.java
package Trees.Model;

public class BinaryTree {
    private Node root;


    private Node insertRecursive(Node current, int value){
        if(current == null){
            return new Node(value);
        }
        if(value <= current.getData()){
            current.setLeft(insertRecursive(current.getLeft(), value));
        }
        else if(value > current.getData()){
            current.setRight(insertRecursive(current.getRight(), value));
        }
        else{
            return current;
        }
        return current;
    }

    public void insert(int value){
        root = insertRecursive(root, value);
    }

    private boolean containsRecursive(Node current, int value){
        if(value == current.getData()){
            return true;
        }
        else if(value < current.getData()){
            if(current.getLeft() == null){
                return false;
            }
            else{
                return this.containsRecursive(current.getLeft(), value);
            }
        }
        else{
            if(current.getRight() == null){
                return false;
            }
            else{
                return this.containsRecursive(current.getRight(), value);
            }
        }
    }

    public boolean contains(int value){
        return containsRecursive(root, value);
    }

    private void printInOrderRecursive(Node current){
        if(current.getLeft() != null){
            printInOrderRecursive(current.getLeft());
        }
        System.out.println(current.getData());
        if(current.getRight() != null){
            printInOrderRecursive(current.getRight());
        }
    }

    public void printInOrder(){
        printInOrderRecursive(root);
    }
}

File: Javascript/BinaryTree/Model/BinaryTree.js
import { BinaryTreeNode } from "./BinaryTreeNode.js";

export class BinaryTree{
    #root;

    constructor(){
        this.#root = null;
    }

    getRoot(){
        return this.#root;
    }

    setRoot(binaryTreeNode){
        this.#root = binaryTreeNode;
    }

    insertRecursive(currentNode, number){
        if(currentNode == null){
            const newBinaryTreeNode = new BinaryTreeNode(number);
            return newBinaryTreeNode;
        }

        if(currentNode.getValue() == number){
            console.log("Value already exist.")
            return currentNode;
        }

        if(number < currentNode.getValue()){
            currentNode.setLeft(this.insertRecursive(currentNode.getLeft(), number));
        }
        else if(number > currentNode.getValue()){
            currentNode.setRight(this.insertRecursive(currentNode.getRight(), number));
        }
        return currentNode;
    }

    insert(number){
        this.#root = this.insertRecursive(this.getRoot(), number);
    }

    print(){
        this.#printInOrder(this.getRoot());
    }

    #printInOrder(currentNode){
        if(currentNode.getLeft() != null){
            this.#printInOrder(currentNode.getLeft());
        }

        console.log(currentNode.getValue());

        if(currentNode.getRight() != null){
            this.#printInOrder(currentNode.getRight());
        }
    }
}